feat: introduce Grid List component

Betrozov opened the pull request:
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/867

## Description
Introducing the new Grid List component

## Screenshots
![grid-list](https://user-images.githubusercontent.com/8318592/97967207-96298b80-1dc5-11eb-81e6-aae29e37971f.png)

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated

jbadan commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+    box-shadow: $fd-grid-list-item-box-shadow;
+    text-align: center;
+
+    h6 {
```
Can you remove all the element selectors?

InnaAtanasova replied:
+1

Betrozov replied:
fixed

jbadan commented on src/theming/sap_fiori_3_dark.scss:
```diff
@@ -102,4 +102,7 @@
   --fdCalendar_Active_Item_Border: none;
+
+  /* Grid List */
+  --fdGrid_List_SingleSelect_Hover_BoxShadow: 0 0.125rem 0.5rem 0 fade(var(--sapContent_ShadowColor), 30);
```
`fade()` is a less function right? If it is, let's get rid of it - we can make a new variable with the output of `fade(var(--sapContent_ShadowColor), 30);`. The designers are supposed to be removing all less functions from the design specs.

InnaAtanasova replied:
+1 
You can contact @jessmetivier and she will handle the communication with the designers.

Betrozov replied:
Asked @jessmetivier to help
thanks for the comment :slightly_smiling_face:

JKMarkowski commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+
+  @include fd-reset();
+
+  display: flex;
+  flex-direction: column;
```
You can use `@include flex(column(` there

Betrozov replied:
fixed

JKMarkowski commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+      &-body {
+        padding: 1rem;
+        height: 100%;
+        display: flex;
+        flex-direction: column;
```
You can use `@include flex(column)` there

Betrozov replied:
fixed

JKMarkowski reviewed (changes requested):
These styles don't match to newest specifications.

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+  display: flex;
+  flex-direction: column;
+
+  .fd-list {
```
Should be `#{$fd-namespace}` instead of `fd`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+  flex-direction: column;
+
+  .fd-list {
+    &__item {
```
You are not supposed to modify the properties of the List directly here. You can create a new class and rely on the css specificity to overwrite the properties of the list-item. For example:
`div class="fd-list__item fd-grid-list__item"....</div>`

Then in `.fd-grid-list__item` you overwrite the properties coming from `fd-list__item`. This will also eliminate the need of using !important, which is not recommended.

Betrozov replied:
already not use `fd-list`

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+      align-items: normal !important;
+      height: 100% !important;
+
+      .fd-toolbar {
+        padding: 0 1rem;
+        align-items: center;
+      }
+
+      &-body {
```
same for these 2 classes. Create additional classes that will overwrite the "default" ones

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+        color: var(--sapContent_MarkerTextColor);
+      }
+
+      .fd-checkbox + .fd-checkbox__label,
+      .fd-radio + .fd-radio__label {
+        display: none;
+        padding: 0 !important;
+        margin: 0 !important;
+      }
+
+      .fd-radio + .fd-radio__label::after {
```
Same as above. And for future reference, if you need to bring `.fd-xxxx` you need to use `#{$fd-namespace}`. If in future we decide to change the prefix we can do it from 1 place.

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+      padding: 0;
+
+      @include fd-active() {
+        * {
+          color: $fd-grid-list-active-text-color !important;
+        }
```
That's not a good idea :)

Betrozov replied:
Do you know another way to avoid using * if we need to apply color for all inside an item and namely inside the body where we do not know what will be inside?
I will be glad for your help @InnaAtanasova :slightly_smiling_face:

JKMarkowski commented on src/list-grid.scss:
```diff
@@ -0,0 +1,232 @@
+  display: flex;
+  flex-direction: column;
+
+  .fd-list {
```
I think you should consider getting rid of `fd-list` usage. The only things that are taken from list are interractive states (hover/active/select). I know that it's named grid-**list**. But now you overwrite almost every element, which will force users to put 2 classes for 1 element (`fd-list__x` and `fd-grid-list__x`), For me you can export interractive states to some mixin(just to keep same for list and grid list) and treat grid list as a standalone component. Let me know what you think about it, or for some more clarifications

InnaAtanasova replied:
+1

Betrozov replied:
agree with you
but about to move interactive states to mixin I think it needs to be done by separate PR
1 PR = 1 component

InnaAtanasova commented:
"Visual" issues:
1. On hover and selected states the inside content covers the rounded borders:
<img width="1464" alt="Screen Shot 2020-11-04 at 9 07 28 AM" src="https://user-images.githubusercontent.com/39598672/98121541-6b9b0980-1e7d-11eb-896e-af26c039af38.png">
<img width="792" alt="Screen Shot 2020-11-04 at 9 09 46 AM" src="https://user-images.githubusercontent.com/39598672/98121594-81103380-1e7d-11eb-9ac8-4144fd2cfe19.png">

2. RTL issues:
<img width="1942" alt="Screen Shot 2020-11-04 at 9 05 04 AM" src="https://user-images.githubusercontent.com/39598672/98121806-c92f5600-1e7d-11eb-87b3-b2022c63f4ca.png">

Betrozov commented:
> RTL issues:
> 
> <img alt="Screen Shot 2020-11-04 at 9 05 04 AM" width="1942" src="https://user-images.githubusercontent.com/39598672/98121806-c92f5600-1e7d-11eb-87b3-b2022c63f4ca.png">

Hi @InnaAtanasova 
In the fd-grid-list__item-body can be whatever developer wants and I think for styles in the body will responsible developer
What do you think?

InnaAtanasova commented:
> > RTL issues:
> > <img alt="Screen Shot 2020-11-04 at 9 05 04 AM" width="1942" src="https://user-images.githubusercontent.com/39598672/98121806-c92f5600-1e7d-11eb-87b3-b2022c63f4ca.png">
> 
> Hi @InnaAtanasova
> In the fd-grid-list__item-body can be whatever developer wants and I think for styles in the body will responsible developer
> What do you think?

True, but the documentation (examples) should not look "broken".

katekozlowska reviewed (changes requested):
with`.fd-list__group-header` comes `padding: 0 1rem,` and it breaks the layout - header should be aligned with a box (first` <li> `element) below

current view:
![Screenshot from 2020-11-04 20-02-03](https://user-images.githubusercontent.com/7014311/98158847-e4c94980-1edb-11eb-8080-301d56b92819.png)

![Screenshot from 2020-11-04 20-05-13](https://user-images.githubusercontent.com/7014311/98158904-f9a5dd00-1edb-11eb-8e92-4dfccffbe0ab.png)

design:
![Screenshot from 2020-11-04 20-02-03 (1)](https://user-images.githubusercontent.com/7014311/98158892-f4489280-1edb-11eb-8232-ef4eed464b7f.png)

Betrozov commented:
> "Visual" issues:
> 
> 1. On hover and selected states the inside content covers the rounded borders:
> 
> <img alt="Screen Shot 2020-11-04 at 9 07 28 AM" width="1464" src="https://user-images.githubusercontent.com/39598672/98121541-6b9b0980-1e7d-11eb-896e-af26c039af38.png">
> 
> No rounded borders + focus line issue
> <img alt="Screen Shot 2020-11-04 at 9 09 46 AM" width="792" src="https://user-images.githubusercontent.com/39598672/98121594-81103380-1e7d-11eb-9ac8-4144fd2cfe19.png">

fixed

> 1. RTL issues:
> 
> <img alt="Screen Shot 2020-11-04 at 9 05 04 AM" width="1942" src="https://user-images.githubusercontent.com/39598672/98121806-c92f5600-1e7d-11eb-87b3-b2022c63f4ca.png">

fixed

Betrozov commented:
> with`.fd-list__group-header` comes `padding: 0 1rem,` and it breaks the layout - header should be aligned with a box (first`<li>`element) below
> 
> current view:
> ![Screenshot from 2020-11-04 20-02-03](https://user-images.githubusercontent.com/7014311/98158847-e4c94980-1edb-11eb-8080-301d56b92819.png)
> 
> (see padding)
> ![Screenshot from 2020-11-04 20-05-13](https://user-images.githubusercontent.com/7014311/98158904-f9a5dd00-1edb-11eb-8e92-4dfccffbe0ab.png)
> 
> design:
> ![Screenshot from 2020-11-04 20-02-03 (1)](https://user-images.githubusercontent.com/7014311/98158892-f4489280-1edb-11eb-8232-ef4eed464b7f.png)

fixed
thanks for the comment

jbadan commented on .storybook/custom/custom.scss:
```diff
@@ -491,3 +491,27 @@ body.sb-show-main {
   border-radius: 0.25rem;
 }
+
+.fd-grid-list__item--body {
```
There shouldn't be any `fd-` classes in this file - just playground classes. That way the styles don't get confusing whether it's part of the actual component or just for the documentation.

Betrozov replied:
fixed

jbadan commented on src/list-grid.scss:
```diff
@@ -0,0 +1,234 @@
+    height: 100%;
+    position: relative;
+
+    .#{$fd-namespace}-toolbar {
```
We should not be using `#{$fd-namespace}` in any file. There should only be 1 `#{$fd-namespace}` class per file. The reason for this is that it 1) violates self contained styles 2) it messes up css modules in fundamental-react

You can change this to `&__toolbar {` and add `fd-grid-list__toolbar` to toolbars in the examples.

Betrozov replied:
fixed

jbadan commented on src/list-grid.scss:
```diff
@@ -0,0 +1,234 @@
+      padding: 0 !important;
+    }
+
+    .#{$block}__radio--label::after {
```
This should work find as `&__radio--label::after {` without the `#{$block}`

Betrozov replied:
Hi @jbadan 
if I make like you want `.#{$block}__radio--label` then I need to use `!important` to avoid that I just made a higher priority 
selector

jbadan replied:
oh, I didn't realize! Thank you for explaining 😃

jbadan commented on src/list-grid.scss:
```diff
@@ -0,0 +1,234 @@
+    box-shadow: $fd-grid-list-item-box-shadow;
+    text-align: center;
+
+    .#{$more}--title {
```
Can be `&--title`

Betrozov replied:
the same answer

jbadan commented on src/list-grid.scss:
```diff
@@ -0,0 +1,234 @@
+    }
+
+    .#{$more}--title,
+    .#{$more}--description {
```
Can be `&--title` and `&--description`

Betrozov replied:
the same answer `&--description`

jbadan reviewed (changes requested):
See comments

Betrozov commented on src/list-grid.scss:
```diff
@@ -0,0 +1,234 @@
+      font-weight: bold;
+    }
+
+    .#{$more}--description {
```
fixed

JKMarkowski commented on src/list-grid.scss:
```diff
@@ -0,0 +1,234 @@
+      color: var(--sapContent_MarkerTextColor);
+    }
+
+    .#{$block}__checkbox--label,
```
![image](https://user-images.githubusercontent.com/26483208/98676185-91cc1800-235b-11eb-9556-174d9449b16a.png)
Focus in checkbox inside grid list looks weird

Betrozov replied:
fixed

JKMarkowski reviewed (changes requested):
Great Job! This PR looks much better, after getting rid of `list` reference. I added 2 minor things, let me know when it's addressed, so I will re-review

Betrozov commented on src/list-grid.scss:
```diff
@@ -0,0 +1,234 @@
+    @include fd-flex-vertical-center();
+
+    padding: 0 1rem;
+  }
```
fixed

JKMarkowski commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+          right: 0;
+          bottom: 0;
+          left: 0;
+          margin: auto !important;
```
I'm not fan of importants in library, but for this case this is for sure better, than creating additional 2-3 more nesting levels

JKMarkowski commented:
@Betrozov 
Can you change `fd-col-lg--3` in all examples to `fd-col-lg--4`? 
![image](https://user-images.githubusercontent.com/26483208/98922438-be567000-24d2-11eb-8f1f-66251a3d4518.png)

Betrozov commented:
> @Betrozov
> Can you change `fd-col-lg--3` in all examples to `fd-col-lg--4`?
> ![image](https://user-images.githubusercontent.com/26483208/98922438-be567000-24d2-11eb-8f1f-66251a3d4518.png)

changed

InnaAtanasova commented:
<img width="1212" alt="Screen Shot 2020-11-12 at 11 56 34 AM" src="https://user-images.githubusercontent.com/39598672/98970621-252d5680-24de-11eb-9930-d42d040e57d3.png">

In this example the cursor is of type pointer on hover over all items. Shouldn't it be pointer only when it's over a clickable item?

InnaAtanasova commented:
<img width="1114" alt="Screen Shot 2020-11-12 at 11 58 48 AM" src="https://user-images.githubusercontent.com/39598672/98971035-a4228f00-24de-11eb-8415-c12678609556.png">

RTL issue with the Radio button. Please check if it's in the scope of this PR and if not, please log an issue. Thanks

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+  @include fd-flex(column);
+
+  &__item {
+    @include fd-flex(column);
```
I think it's missing a reset

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    height: 100%;
+    position: relative;
+
+    &--body {
```
This is not a modifier class. You are using it like this:
<img width="496" alt="Screen Shot 2020-11-12 at 12 07 32 PM" src="https://user-images.githubusercontent.com/39598672/98971850-b3eea300-24df-11eb-9b14-023b13578567.png">

It's a separate div element, it's not modifying the properties of the `fd-grid-list__item--body`. You can name it `fd-grid-list__item-body` and move it out of the `fd-grid-list__item` block. It will also need a reset.

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    position: relative;
+
+    &--body {
+      @include fd-flex(column);
```
```
@include fd-flex(column) {
    justify-content: space-between;
}
```

You can use it like this. Keeps everything related to flex in one place.

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+      }
+    }
+
+    &--counter {
```
<img width="419" alt="Screen Shot 2020-11-12 at 12 11 39 PM" src="https://user-images.githubusercontent.com/39598672/98972307-40996100-24e0-11eb-9603-fa87574d289d.png">

This is also not a modifier class. It would be a modifier class if you had `fd-grid-list__item fd-grid-list__item--counter`, then `fd-grid-list__item--counter` will modify the properties coming from `fd-grid-list__item`.
In your case it's a separate element and can be named `fd-grid-list__item-counter` or `fd-grid-list__counter` for simplicity. And you will have to add a reset mixin as it's a standalone element. Only modifier classes don't have reset, as the main class that they are modifying is bringing the reset.

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+      color: var(--sapContent_MarkerTextColor);
+    }
+
+    .#{$block}__radio--input {
```
`.#{$block}__radio-input` or `.#{$block}__radio`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+        }
+      }
+
+      + .#{$block}__radio--label {
```
`.#{$block}__radio-label`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+      }
+    }
+
+    .#{$block}__checkbox--input {
```
`.#{$block}__checkbox-input` or `.#{$block}__checkbox`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+
+    .#{$block}__checkbox--input {
+      @include fd-focus() {
+        + .#{$block}__checkbox--label {
```
`.#{$block}__checkbox-label`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+        }
+      }
+
+      + .#{$block}__checkbox--label {
```
`.#{$block}__checkbox-label`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    }
+
+    .#{$block}__checkbox--label,
+    .#{$block}__radio--label {
```
`.#{$block}__radio-label`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+      display: none;
+    }
+
+    &--state-unread {
```
can be `&--unread`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+      }
+    }
+
+    &--state-actions {
```
can be `&--actions`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    padding: 0 1rem;
+  }
+
+  .#{$block}__group-header {
```
needs a reset?

InnaAtanasova replied:
although the way it is the rest might break the styling coming from `fd-col`. Are you sure `fd-col fd-col--12 fd-grid-list__group-header` has to be together? In other cases you add the `fd-col` classes on a separate wrapper div.

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    }
+  }
+
+  &__mode {
```
<img width="474" alt="Screen Shot 2020-11-12 at 12 22 42 PM" src="https://user-images.githubusercontent.com/39598672/98973464-cc5fbd00-24e1-11eb-992e-7ccda8aca42f.png">

You can't have a structure like this. Following BEM:
`fd-grid-list` is the **Block**. You can't have a **Block** AND an **Element** `fd-grid-list__mode` on the same level, meaning you can't have `fd-grid-list fd-grid-list__mode`. The **Element** (the one that you add `__` in the name) is a child of the Block. The modifier (`--` in the name) is the one you can use with Blocks and Elements. So you can have `fd-grid-list fd-grid-list--dark-mode` for example, but you can't have `fd-grid-list fd-grid-list__mode`.

InnaAtanasova replied:
And you will need a reset if it's an **Element**

Betrozov replied:
fixed

Betrozov replied:
thank you for the comment :slightly_smiling_face: 
@InnaAtanasova

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+      }
+    }
+
+    &--single-select-master,
```
Starting Jan 1st we need to get abandon the usage of words like `master`, `slave`, etc. Could you find another name for this class?

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    }
+  }
+
+  &__more {
```
This needs a reset

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    box-shadow: $fd-grid-list-item-box-shadow;
+    text-align: center;
+
+    .#{$more}--title {
```
This is not a modifier class. It should be named `fd-grid-list__title` or `fd-grid-list__more-title`

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+      font-weight: bold;
+    }
+
+    &--description {
```
same here. It should be names `fd-grid-list__more-description` and since it's a standalone it needs a reset

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+    }
+  }
+
+  &__footer {
```
missing reset

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,261 @@
+  }
+
+  @each $set-name, $colors in $fd-grid-list-highlight {
+    &__highlight--#{$set-name} {
```
You are missing a `fd-grid-list__highlight` class that will bring the reset. You put there all the rules that are shared amongst all states (error, succes, etc). You then create a modifier class `fd-grid-list__highlight--error` that will overwrite the property specific for the base case. In this case it will be the color. The rest of the properties (position, top, left, etc. will go in the base class as they are shared). You need to have something like:
`<span class="fd-grid-list__highlight fd-grid-list__highlight--error">`

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+            <div class="fd-col fd-col--12 fd-col-md--6 fd-col-lg--4 fd-col-xl--3">
+                <div tabindex="0" class="fd-grid-list__item fd-grid-list__item--link">
+                    <input type="radio" class="fd-radio fd-radio--compact fd-grid-list__radio--input" id="pDidh761" name="singleSelectMasterRadio">
+                    <label class="fd-radio__label fd-grid-list__radio--label" for="pDidh761"></label>
```
This should not be an empty label. At the moment when you focus the radio button, the entire card reads out. There should be some sort of label to summarize what the item is (title, etc)

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+                <div tabindex="0" class="fd-grid-list__item fd-grid-list__item--link is-selected">
+                    <div class="fd-toolbar fd-grid-list__toolbar">
+                        <input type="checkbox" class="fd-checkbox fd-checkbox--compact fd-grid-list__checkbox--input" id="Ai4ez611c" checked>
+                        <label class="fd-checkbox__label fd-checkbox__label--compact fd-grid-list__checkbox--label" for="Ai4ez611c"></label>
```
Same feedback as for the radio buttons, except for some reason the content of the grid item doesn't read out when you access this checkbox, just 'checkbox'. Every input needs some sort of label.

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+
+        <div class="fd-row">
+            <div  class="fd-col fd-col--12 fd-grid-list__group-header">
+                State: Unread
```
This group header is not announced when you tab into the group. It should be read upon accessing the first grid item. The same applies for all header examples.

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+
+                        <div class="fd-grid-list__item--state-actions">
+                            <span class="fd-object-status fd-object-status--negative">
+                                <span class="fd-object-status__text">(Contains errors)</span>
```
The only problem with this layout is that the error state is not read out until the end of the content. So a screen reader user would not know unless they sat there to listen to the entire card content until it got to this line. Ideally the validation state should be announced first.

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+        <div class="fd-row">
+            <div  class="fd-col fd-col--12 fd-col-md--6 fd-col-lg--4 fd-col-xl--3">
+                <div tabindex="0" class="fd-grid-list__item">
+                    <span class="fd-grid-list__highlight--success"></span>
```
There needs to be some screen reader support for the highlight states so screen reader users are aware of a) that it's highlighted at all and b) what the color means. 

This should honestly go back to UX since this is using only color to indicate a state and that is not accessible for several user groups, not just the visually impaired.

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+
+        <div class="fd-row">
+            <div class="fd-col fd-col--12">
+                <div class="fd-toolbar fd-toolbar--info fd-toolbar--active">
```
The content of this div is not announced to screen readers. The button is accessible, but simply announces 'Cancel' without context.

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+
+        <div class="fd-row">
+            <div class="fd-col fd-col--12">
+                <div class="fd-grid-list__more">
```
This section is not accessible by keyboard.

Betrozov replied:
fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,1880 @@
+        <div class="fd-row">
+            <div class="fd-col fd-col--12">
+                <div class="fd-grid-list__footer">
+                    This is footer text
```
This is not accessible to screen reader users.

Betrozov replied:
fixed
thank you for the comments @meganaconley

Betrozov commented:
> <img alt="Screen Shot 2020-11-12 at 11 56 34 AM" width="1212" src="https://user-images.githubusercontent.com/39598672/98970621-252d5680-24de-11eb-9930-d42d040e57d3.png">
> 
> In this example the cursor is of type pointer on hover over all items. Shouldn't it be pointer only when it's over a clickable item?

@InnaAtanasova It is a single select mode we can select only one item from all items in the list that's why we need cursor is of type pointer here

InnaAtanasova commented:
> > <img alt="Screen Shot 2020-11-12 at 11 56 34 AM" width="1212" src="https://user-images.githubusercontent.com/39598672/98970621-252d5680-24de-11eb-9930-d42d040e57d3.png">
> > In this example the cursor is of type pointer on hover over all items. Shouldn't it be pointer only when it's over a clickable item?
> 
> @InnaAtanasova It is a single select mode we can select only one item from all items in the list that's why we need cursor is of type pointer here

In this case why the cursor is not of type pointer when you hover over the selected item?

InnaAtanasova commented:
Shouldn't the `delete` button be placed before the `navigation indicator`?
How it's now:
<img width="297" alt="Screen Shot 2020-11-13 at 7 22 19 AM" src="https://user-images.githubusercontent.com/39598672/99071962-1f3c8180-2581-11eb-91b3-88d3f37e0d9c.png">

In specs:
<img width="356" alt="Screen Shot 2020-11-13 at 7 22 25 AM" src="https://user-images.githubusercontent.com/39598672/99071974-25326280-2581-11eb-8c0b-bc07a9c5511a.png">

InnaAtanasova commented:
I think you are missing an example in the documentation for `GridList with mode SingleSelect` where the radio buttons are displayed on the right hand side:
<img width="842" alt="Screen Shot 2020-11-13 at 7 25 40 AM" src="https://user-images.githubusercontent.com/39598672/99072206-92de8e80-2581-11eb-97bf-c7bad9b66fc3.png">

Betrozov commented:
> <img alt="Screen Shot 2020-11-12 at 11 58 48 AM" width="1114" src="https://user-images.githubusercontent.com/39598672/98971035-a4228f00-24de-11eb-8415-c12678609556.png">
> 
> RTL issue with the Radio button. Please check if it's in the scope of this PR and if not, please log an issue. Thanks

fixed

Betrozov commented:
> > > <img alt="Screen Shot 2020-11-12 at 11 56 34 AM" width="1212" src="https://user-images.githubusercontent.com/39598672/98970621-252d5680-24de-11eb-9930-d42d040e57d3.png">
> > > In this example the cursor is of type pointer on hover over all items. Shouldn't it be pointer only when it's over a clickable item?
> > 
> > 
> > @InnaAtanasova It is a single select mode we can select only one item from all items in the list that's why we need cursor is of type pointer here
> 
> In this case why the cursor is not of type pointer when you hover over the selected item?

fixed

Betrozov commented:
> Shouldn't the `delete` button be placed before the `navigation indicator`?
> How it's now:
> <img alt="Screen Shot 2020-11-13 at 7 22 19 AM" width="297" src="https://user-images.githubusercontent.com/39598672/99071962-1f3c8180-2581-11eb-91b3-88d3f37e0d9c.png">
> 
> In specs:
> <img alt="Screen Shot 2020-11-13 at 7 22 25 AM" width="356" src="https://user-images.githubusercontent.com/39598672/99071974-25326280-2581-11eb-8c0b-bc07a9c5511a.png">

fixed

Betrozov commented:
> I think you are missing an example in the documentation for `GridList with mode SingleSelect` where the radio buttons are displayed on the right hand side:
> <img alt="Screen Shot 2020-11-13 at 7 25 40 AM" width="842" src="https://user-images.githubusercontent.com/39598672/99072206-92de8e80-2581-11eb-97bf-c7bad9b66fc3.png">

added

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,301 @@
+    height: 100%;
+    padding: 1rem;
+
+    &---no-padding {
```
Should be 2 `--` not 3

Betrozov replied:
fixed

InnaAtanasova commented on src/list-grid.scss:
```diff
@@ -0,0 +1,301 @@
+    color: var(--sapContent_MarkerTextColor);
+  }
+
+  .#{$block}__toolbar {
```
should be renamed to `__item-toolbar` as it's part of the item

Betrozov replied:
fixed

InnaAtanasova commented:
Sorry, one last thing for the RTL:
<img width="298" alt="Screen Shot 2020-11-18 at 8 43 21 AM" src="https://user-images.githubusercontent.com/39598672/99543231-f5200080-2980-11eb-815b-6b72a9e1981d.png">
<img width="301" alt="Screen Shot 2020-11-18 at 8 43 13 AM" src="https://user-images.githubusercontent.com/39598672/99543233-f5b89700-2980-11eb-9b89-709b95842468.png">

You just need to move the margin to the other side on RTL
<img width="299" alt="Screen Shot 2020-11-18 at 8 43 47 AM" src="https://user-images.githubusercontent.com/39598672/99543291-06690d00-2981-11eb-8ed2-9296f7e9c17c.png">

Betrozov commented:
> Sorry, one last thing for the RTL:
> <img alt="Screen Shot 2020-11-18 at 8 43 21 AM" width="298" src="https://user-images.githubusercontent.com/39598672/99543231-f5200080-2980-11eb-815b-6b72a9e1981d.png">
> <img alt="Screen Shot 2020-11-18 at 8 43 13 AM" width="301" src="https://user-images.githubusercontent.com/39598672/99543233-f5b89700-2980-11eb-9b89-709b95842468.png">
> 
> You just need to move the margin to the other side on RTL
> <img alt="Screen Shot 2020-11-18 at 8 43 47 AM" width="299" src="https://user-images.githubusercontent.com/39598672/99543291-06690d00-2981-11eb-8ed2-9296f7e9c17c.png">

fixed

meganaconley commented on stories/list-grid/list-grid.stories.js:
```diff
@@ -0,0 +1,2069 @@
+
+        <div class="fd-row">
+            <div class="fd-col fd-col--12">
+                <div tabindex="0" class="fd-grid-list__more">
```
Sorry I missed this initially, but this should have `role="button"` as well if it can't be changed to be a true button.

Betrozov replied:
fixed))

meganaconley reviewed (changes requested):
Great work on the accessibility feedback. I really like the callouts for the highlighted items and error state.

I think there were some regressions though. The focus ring for the checkbox/radio buttons is off center. There is an invisible focusable radio button on the single select cards as well. The focus for that one should be just around the card, and using arrow keys should navigate between them. I believed these looked correct last time I reviewed.

<img width="1021" alt="Screen Shot 2020-11-18 at 11 52 28 AM" src="https://user-images.githubusercontent.com/57502177/99581174-40440e80-2995-11eb-920a-abc1c69fbc78.png">

<img width="1056" alt="Screen Shot 2020-11-18 at 11 53 10 AM" src="https://user-images.githubusercontent.com/57502177/99581191-476b1c80-2995-11eb-946a-46c3d2981876.png">

<img width="1020" alt="Screen Shot 2020-11-18 at 11 53 36 AM" src="https://user-images.githubusercontent.com/57502177/99581200-49cd7680-2995-11eb-8765-a18c6ba4a88c.png">

Betrozov commented:
> Great work on the accessibility feedback. I really like the callouts for the highlighted items and error state.
> 
> I think there were some regressions though. The focus ring for the checkbox/radio buttons is off center. There is an invisible focusable radio button on the single select cards as well. The focus for that one should be just around the card, and using arrow keys should navigate between them. I believed these looked correct last time I reviewed.
> 
> <img alt="Screen Shot 2020-11-18 at 11 52 28 AM" width="1021" src="https://user-images.githubusercontent.com/57502177/99581174-40440e80-2995-11eb-920a-abc1c69fbc78.png">
> 
> <img alt="Screen Shot 2020-11-18 at 11 53 10 AM" width="1056" src="https://user-images.githubusercontent.com/57502177/99581191-476b1c80-2995-11eb-946a-46c3d2981876.png">
> 
> <img alt="Screen Shot 2020-11-18 at 11 53 36 AM" width="1020" src="https://user-images.githubusercontent.com/57502177/99581200-49cd7680-2995-11eb-8765-a18c6ba4a88c.png">

fixed
